### DISCUSSION TITLE
Xdm MDX compiler

### DISCUSSION THREAD

**Discussion by rsipakov** [timlrx/tailwind-nextjs-starter-blog#107] [Q&A] [answered]
I try switching the MDX bundler from "next-mdx-remote" to "mdx-bundler" following these steps: https://github.com/timlrx/tailwind-nextjs-starter-blog/commit/553722750e65b9c9b640191f38f82a82036eec17 and https://github.com/timlrx/tailwind-nextjs-starter-blog/commit/f7cda6f536705993b2e6b7953c21372ae4b0fcad but I got an error:

`Server Error
TypeError: (0 , _utils_files__WEBPACK_IMPORTED_MODULE_8__.default) is not a function`

```
lib/mdx.js (147:39) @ getAllFilesFrontMatter

  145 |   const prefixPaths = path.join(root, 'data', folder)
  146 | 
> 147 |   const files = getAllFilesRecursively(prefixPaths)
      |                                       ^
  148 | 
  149 |   const allFrontMatter = []
  150 | 
```

```
Call Stack
getStaticProps
pages/index.js (22:45)
Show collapsed frames
```

Could you help me resolve this issue? Thank you!

**Comment by rsipakov** [comment]
This issue was resolved by update files.js (ESLint auto-fix made incorrect changes into this file). But when I try open the blog post page then I got another issue with the plugin: esbuild-xdm

```
Error: Build failed with 1 error:
components/_mdx_bundler_entry_point.mdx:82:163: error: [plugin: esbuild-xdm] Could not parse expression with acorn: Unexpected token
```

```
errors: [
    {
      detail: [Error],
      location: [Object],
      notes: [],
      pluginName: 'esbuild-xdm',
      text: 'Could not parse expression with acorn: Unexpected token'
    }
  ],
  warnings: []
}
```

**Reply by timlrx** [reply]
Does it work with just a simple mdx file like those in the `v1` branch? There are some changes that need to be made to mdx files e.g. https://github.com/timlrx/tailwind-nextjs-starter-blog/commit/553722750e65b9c9b640191f38f82a82036eec17#diff-af540c34202b4bfc8a9f7426aba91d4547b804d945747b5db5665f1554ab95daL56

The setup is also relatively complicated so if you clone and install a fresh branch and patch your changes incrementally, it might be easier to debug what's the issue.

**Comment by rsipakov** [comment, marked as answer]
@timlrx Thank you very much for your advice. Yes, I used {...} in the .mdx file. After removed {...} it works fine.
